src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -1896,6 +1909,51 @@ export class SavedObjectsRepository {
     return body;
   }
 
+  /**
+   * Returns a generator to help page through large sets of saved objects.
+   *
+   * The generator wraps calls to `SavedObjects.find` and iterates over
+   * multiple pages of results using `_pit` and `search_after`. This will
+   * open a new Point In Time (PIT), and continue paging until a set of
+   * results is received that's smaller than the designated `perPage`.
```

Comment:
Because the IPointInTimeFinder has useful docs maybe we should link to that. Something like

```
   * Returns a {@link IPointInTimeFinder} to help page through large sets of saved objects.
 ```

Reply:
It might be helpful to give some guidance to developers for when they should use this API: "We strongly recommend using this API for any find queries that might return more than a 1000 saved objects"

I think the only use case the pointInTimeFinder doesn't solve is if you want to open a PIT and do multiple find requests against it before closing it. Maybe we can add a comment to the raw open/close PIT API's like "Only use this API if you have an advanced use case that's not solved by the `createPointInTimeFinder` method"

Reply:
Good idea, I've pushed a handful of docs updates.